### PR TITLE
Adaptivity fixes

### PR DESCRIPTION
**Pull request by gpolitis**

### CODE REVIEW

File: src/org/jitsi/impl/neomedia/rtp/RTPEncodingDesc.java [hunk 1]
```diff
@@ -120,6 +125,35 @@ public FrameDesc put(Long key, FrameDesc value)
...
+        = new TreeMap<Long, FrameDesc>()
+    {
+        /**
+         * A helper {@link LinkedList} that is used to cleanup the map.
+         */
+        private LinkedList<Long> tsl = new LinkedList<>();
+
+        /**
+         * {@inheritDoc}
+         */
```
Comment (bbaldino) [inline src/org/jitsi/impl/neomedia/rtp/RTPEncodingDesc.java]: this must be doing something extra than the parent, right? some extra docs here describing what extra logic it has would be good

File: src/org/jitsi/impl/neomedia/rtp/RTPEncodingDesc.java [hunk 2]
```diff
@@ -176,15 +210,20 @@ public RTPEncodingDesc(
...
      */
     public RTPEncodingDesc(
         MediaStreamTrackDesc track, int idx,
         long primarySSRC, long rtxSSRC,
         int tid, int sid,
+        int resolution,
         RTPEncodingDesc[] dependencyEncodings)
     {
+        // XXX we should be able to snif the actual resolution from the RTP
+        // packets.
```
Comment (bbaldino) [inline src/org/jitsi/impl/neomedia/rtp/RTPEncodingDesc.java]: how would we do that?

File: src/org/jitsi/impl/neomedia/rtp/RTPEncodingDesc.java [hunk 3]
```diff
@@ -161,7 +195,7 @@ public RTPEncodingDesc(MediaStreamTrackDesc track, long primarySSRC)
     public RTPEncodingDesc(
         MediaStreamTrackDesc track, long primarySSRC, long rtxSSRC)
     {
-        this(track, 0, primarySSRC, rtxSSRC,
+        this(track, 0, primarySSRC, rtxSSRC, -1 /* resolution */,
```
Comment (bbaldino) [inline src/org/jitsi/impl/neomedia/rtp/RTPEncodingDesc.java]: if `-1` is the base/invalid resolution, i think a constant would be nice

File: src/org/jitsi/impl/neomedia/rtp/RTPEncodingDesc.java (hunk 2 above)
Comment (gpolitis) [inline src/org/jitsi/impl/neomedia/rtp/RTPEncodingDesc.java]: We could do this by inspecting the payload (of keyframes). This wouldn't work with PERC tho, in which case we should get it from signaling.

**Comment by gpolitis** [comment]
@bbaldino thanks very much for the review and all your suggestions 👍 Let's wait for @bgrozev to have a say.

**Comment by gpolitis** [comment]
@bbaldino I talked to @bgrozev and he's done with the review. Could you please merge this so that I can update the JVB pom file?

**Pull request merged**